### PR TITLE
Added loops.md in kotlin folder

### LINKED ISSUE #1387: [Concept Entry] Kotlin: Loops
### Reason/inspiration (optional)

We would like a new concept entry for [Kotlin](https://www.codecademy.com/resources/docs/kotlin) about loops.

The entry should go in a new file in docs/content/kotlin/concepts/loops/loops.md

The entry should:

- Have a section on each type of looping control structure..
- Have a syntax pseudo block for each type of loop.
- Have an example code block for each loop

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by dakshdeepHERE**
### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

Fixes #1387

### CODE REVIEW

**Comment by dakshdeepHERE** [comment]
@SSwiniarski I deleted my old forked repo as the problem was that I committed my arrays changes in my main branch only thats why whenever I was making a difference branch and pushing the changes the commits were pushing too and leading to that CLA errors. Now I will make sure I don't commit to main branch of my fork mistakenly.

**Comment by dakshdeepHERE** [comment]
image

I ran yarn format:all and yarn lint in my terminal, and there were no errors.

File: content/kotlin/concepts/loops/loops.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+---
+Title: 'Loops'
+Description: 'In kotlin, a loop is a control structure that allows you to repeat a block of code a specified number of times or until a certain condition is met. Loops are useful for automating repetitive tasks and for processing large amounts of data efficiently.'
```
Comment (SSwiniarski) [inline content/kotlin/concepts/loops/loops.md:3]:
```suggestion
Description: 'A loop is a control structure that allows you to repeat a block of code a specified number of times or until a certain condition is met.'
```

I'd move "Loops are useful for automating repetitive tasks and for processing large amounts of data efficiently." into the opening paragraph.

File: content/kotlin/concepts/loops/loops.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'loop'
+  - 'while'
+  - 'for'
+  - 'do-while'
```
Comment (SSwiniarski) [inline content/kotlin/concepts/loops/loops.md:10]:
```suggestion
```

"do-while" is not listed in the tags.md document.

File: content/kotlin/concepts/loops/loops.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
+---
+Title: 'Loops'
+Description: 'In kotlin, a loop is a control structure that allows you to repeat a block of code a specified number of times or until a certain condition is met. Loops are useful for automating repetitive tasks and for processing large amounts of data efficiently.'
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'loop'
```
Comment (SSwiniarski) [inline content/kotlin/concepts/loops/loops.md:7]:
```suggestion
  - 'Loops'
```

File: content/kotlin/concepts/loops/loops.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  - 'while'
```
Comment (SSwiniarski) [inline content/kotlin/concepts/loops/loops.md:8]:
```suggestion
  - 'While'
```

File: content/kotlin/concepts/loops/loops.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  - 'for'
```
Comment (SSwiniarski) [inline content/kotlin/concepts/loops/loops.md:9]:
```suggestion
  - 'For'
```

File: content/kotlin/concepts/loops/loops.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  - 'paths/computer-science'
+---
+
+In Kotlin, a loop is a control flow statement that allows you to repeat a block of code a certain number of times or until a certain condition is met. Kotlin provides several types of loops, including:
+
+- `for` loop: This loop iterates over a range of values or elements in a collection.
+- `while` loop: This loop repeatedly executes a block of code as long as a certain condition is true.
+- `do-while` loop: This loop is similar to a while loop, but it guarantees that the block of code will be executed at least once before the condition is checked.
+
+## While loop
```
Comment (SSwiniarski) [inline content/kotlin/concepts/loops/loops.md:22]: You should have the sections in the same order as they're listed in the bulleted list. You can change one or the other, but the order you have in the list seems the most natural: 1. for, 2. while, 3. do-while

File: content/kotlin/concepts/loops/loops.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+The `while` loops used to execute a block of code repeatedly as long as a certain condition is true. The syntax for a `while` loop in Kotlin is:
```
Comment (SSwiniarski) [inline content/kotlin/concepts/loops/loops.md:24]:
```suggestion
The `while` loop is used to execute a block of code repeatedly as long as a certain condition is true. The syntax for a `while` loop in Kotlin is:
```

File: content/kotlin/concepts/loops/loops.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+```kotlin
```
Comment (SSwiniarski) [inline content/kotlin/concepts/loops/loops.md:26]: This should be a ```pseudo block

File: content/kotlin/concepts/loops/loops.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+```
+1 2 3 4 5 6 7 8 9 10
+```
+
+## For loop
+
+The `for` loop is used to iterate over a range of values or an iterable collection. The syntax for a `for` loop in Kotlin is:
+
+```kotlin
```
Comment (SSwiniarski) [inline content/kotlin/concepts/loops/loops.md:56]: This should be a ```pseudo block.

File: content/kotlin/concepts/loops/loops.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+fun main(){
+    do {
+    // code to be executed at least once
+    } while (condition)
+}
+```
+
+Here's an example of a `do-while` loop in Kotlin that prompts the user to enter a number and prints the decreasing order sequence till 0:
```
Comment (SSwiniarski) [inline content/kotlin/concepts/loops/loops.md:92]: This is not what the example does. Either change the example to match the description, or alter the description.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@dakshdeepHERE I did a first review and left a few comments.

File: content/kotlin/concepts/loops/loops.md (hunk 10 above)
Comment (dakshdeepHERE) [reply content/kotlin/concepts/loops/loops.md:91]: alright wait I will try to fix the description

File: content/kotlin/concepts/loops/loops.md (hunk 6 above)
Comment (dakshdeepHERE) [reply content/kotlin/concepts/loops/loops.md:21]: hmmm alright so I will change the order of this list as the code mentioned below

**Comment by dakshdeepHERE** [comment]
@SSwiniarski check the changes please if it fixed all your requests.

**Comment by dakshdeepHERE** [comment]
Also please @SSwiniarski assign me the issue#1385 on variables as I'm done with my 4 issue's pr.

File: content/kotlin/concepts/loops/loops.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+
+```pseudo
+fun main(){
+    do {
+    // code to be executed at least once
+    } while (condition)
+}
+```
+
+Here is an example of a `do-while` loop in Kotlin. In this code, a variable `i` is declared with an initial value of 10. The loop will continue to run as long as `i` is greater than 0. Inside the loop, the value of `i` is printed and then `i` is decremented by 1 on each iteration. This means that the value of `i` will be printed and reduced by 1 on each iteration until `i` is no longer greater than 0, at which point the loop will exit and the program will end:
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/loops/loops.md:91]: @SSwiniarski let me know if this is right. If you don't like it I can change the example with the description.

**Review by SSwiniarski — APPROVED** [review]
@dakshdeepHERE This looks ready for a second review.

**Comment by dakshdeepHERE** [comment]
@SSwiniarski  alright thank you.

**Comment by dakshdeepHERE** [comment]
@SSwiniarski why my PRs are not getting checked further? Others PR are also getting merged.

**Comment by dakshdeepHERE** [comment]
@yoiang

**Comment by yoiang** [comment]
@dakshdeepHERE i am no longer an active member of this community so i cannot review your PR. 

In the future might i suggest checking who is by using Github's Insights when you ask for reviewers?

**Comment by dakshdeepHERE** [comment]
@yoiang actually I wanted to tag @yangc95

File: content/kotlin/concepts/loops/loops.md [hunk 12]
```diff
@@ -0,0 +1,107 @@
+---
+Title: 'Loops'
+Description: 'A loop is a control structure that allows you to repeat a block of code a specified number of times or until a certain condition is met.'
```
Comment (yangc95) [inline content/kotlin/concepts/loops/loops.md:3]:
```suggestion
Description: 'A loop is a control structure that repeatedly executes a block of code a specified number of times or until a certain condition is met.'
```

File: content/kotlin/concepts/loops/loops.md [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+Tags:
+  - 'Loops'
+  - 'While'
+  - 'For'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+In Kotlin, a loop is a control flow statement that allows you to repeat a block of code a certain number of times or until a certain condition is met. Kotlin provides several types of loops, including:
```
Comment (yangc95) [inline content/kotlin/concepts/loops/loops.md:15]:
```suggestion
In Kotlin, a loop is a control flow statement that allows repeated execution of a block of code for a certain number of times or until a certain condition is met. Kotlin provides several types of loops, including:
```

File: content/kotlin/concepts/loops/loops.md [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+
+- `for` loop: This loop iterates over a range of values or elements in a collection.
```
Comment (yangc95) [inline content/kotlin/concepts/loops/loops.md:17]:
```suggestion
- The `for` loop which iterates over a range of values or elements in a collection.
```

File: content/kotlin/concepts/loops/loops.md [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+- `while` loop: This loop repeatedly executes a block of code as long as a certain condition is true.
```
Comment (yangc95) [inline content/kotlin/concepts/loops/loops.md:18]:
```suggestion
- The `while` loop which repeatedly executes a block of code as long as a certain condition is true.
```

File: content/kotlin/concepts/loops/loops.md [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+- `do-while` loop: This loop is similar to a while loop, but it guarantees that the block of code will be executed at least once before the condition is checked.
```
Comment (yangc95) [inline content/kotlin/concepts/loops/loops.md:19]:
```suggestion
- The `do-while` loop which is similar to a while loop, but it guarantees that the block of code will be executed at least once before the condition is checked.
```

File: content/kotlin/concepts/loops/loops.md [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+
+## For loop
+
+The `for` loop is used to iterate over a range of values or an iterable collection. The syntax for a `for` loop in Kotlin is:
+
+```pseudo
+fun main(){
```
Comment (yangc95) [inline content/kotlin/concepts/loops/loops.md:26]:
```suggestion
fun main() {
```

File: content/kotlin/concepts/loops/loops.md [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+    for (item in collection) {
+   // code to be executed for each item
+    }
```
Comment (yangc95) [inline content/kotlin/concepts/loops/loops.md:29]:
```suggestion
  for (item in collection) {
    // code to be executed for each item
  }
```

File: content/kotlin/concepts/loops/loops.md [hunk 19]
```diff
@@ -0,0 +1,107 @@
...
+}
+```
+
+Here's an example of a `for` loop in Kotlin that prints the numbers from 1 to 10:
+
+```kotlin
+fun main(){
+    for(i in 1..10){
+        print(i)
+    }
```
Comment (yangc95) [inline content/kotlin/concepts/loops/loops.md:39]:
```suggestion
fun main() {
  for(i in 1..10){
    print(i)
  }
```

File: content/kotlin/concepts/loops/loops.md [hunk 20]
```diff
@@ -0,0 +1,107 @@
...
+
+The output for the above code will be:
+
+```
+1 2 3 4 5 6 7 8 9 10
+```
+
+## While loop
+
+The `while` loops used to execute a block of code repeatedly as long as a certain condition is true. The syntax for a `while` loop in Kotlin is:
```
Comment (yangc95) [inline content/kotlin/concepts/loops/loops.md:51]:
```suggestion
The `while` loop is used to execute a block of code repeatedly as long as a certain condition is true. The syntax for a `while` loop in Kotlin is:
```

File: content/kotlin/concepts/loops/loops.md [hunk 21]
```diff
@@ -0,0 +1,107 @@
...
+
+```pseudo
+    fun main(){
+        while(condition){
+            //code to be execute while the condition is true
+        }
+    }
```
Comment (yangc95) [inline content/kotlin/concepts/loops/loops.md:58]:
```suggestion
fun main() {
  while (condition) {
    // Code to be execute while the condition is true
  }
}
```

File: content/kotlin/concepts/loops/loops.md [hunk 22]
```diff
@@ -0,0 +1,107 @@
...
+
+## Do-While loop
+
+A `do-while` loop in Kotlin is similar to a `while` loop, but the block of code is always executed at least once before the condition is checked. The syntax for a `do-while` loop is:
+
+```pseudo
+fun main(){
+    do {
+    // code to be executed at least once
+    } while (condition)
```
Comment (yangc95) [inline content/kotlin/concepts/loops/loops.md:87]: Can we do 2-space indents here?

File: content/kotlin/concepts/loops/loops.md (hunk 11 above)
Comment (yangc95) [inline content/kotlin/concepts/loops/loops.md:91]:
```suggestion
In the example below, a variable `i` is declared with an initial value of 10. The loop will continue to run as long as `i` is greater than 0. Inside the loop, the value of `i` is printed and then `i` is decremented by 1 on each iteration. This means that the value of `i` will be printed and reduced by 1 on each iteration until `i` is no longer greater than 0, at which point the loop will exit and the program will end:
```

File: content/kotlin/concepts/loops/loops.md (hunk 22 above)
Comment (dakshdeepHERE) [reply content/kotlin/concepts/loops/loops.md:87]: @yangc95 I write in intellij see it take default 1 space indent see I will attach ss also down below.
image

File: content/kotlin/concepts/loops/loops.md [hunk 23]
```diff
@@ -0,0 +1,107 @@
...
+
+Here is an example of a `do-while` loop in Kotlin. In this code, a variable `i` is declared with an initial value of 10. The loop will continue to run as long as `i` is greater than 0. Inside the loop, the value of `i` is printed and then `i` is decremented by 1 on each iteration. This means that the value of `i` will be printed and reduced by 1 on each iteration until `i` is no longer greater than 0, at which point the loop will exit and the program will end:
+
+```kotlin
+fun main(){
+    var i = 10
+    do {
+    print(" " + i)
+    i--
+    } while (i > 0)
```
[the remark this answers is not in the document]
Comment (dakshdeepHERE) [reply content/kotlin/concepts/loops/loops.md:99]: I don't think it's wrong

[the remark this answers is not in the document]
Comment (dakshdeepHERE) [reply content/kotlin/concepts/loops/loops.md:99]: wait yes yes we have to shift the print function 1 space ahead. Sorry.

**Comment by dakshdeepHERE** [comment]
@yangc95 please check the comments I made if they are right. Also I have linked a pic for your indent question.

**Comment by dakshdeepHERE** [comment]
@yangc95 can we move forward with this pr.

**Review by yangc95 — APPROVED** [review]
This looks ready for merging now! I made some changes here: [2e20243](https://github.com/Codecademy/docs/pull/1417/commits/2e202435c865f5d40433ae5837ae8c8d2cc30526)&[8f68021](https://github.com/Codecademy/docs/pull/1417/commits/8f6802188ca0092eb4b8d3a1fadbefb81605adc4)&[c197b7f](https://github.com/Codecademy/docs/pull/1417/commits/c197b7febcf712b6b5d32e8f3cff1a812adcc9ec)

**Comment by yangc95** [comment]
🥳 @dakshdeepHERE live link: https://www.codecademy.com/resources/docs/kotlin/loops

Looks good!
Screen Shot 2023-01-27 at 10 32 44 PM

TY for everyone's work on this 💯
cc: @SSwiniarski

**Pull request merged**